utils/nctl/scripts/config_gen.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=line-too-long
+
+import argparse
+import logging
+import pprint as pp
+import sys
+from pathlib import Path
+from tomlkit import dumps
+from tomlkit import loads
+
+
+logger = logging.getLogger(__name__)
+
+
+def main():
+    """The meat and potatoes."""
+    args = parse_args()
+    logger = logger_setup(args.log_level)
+    logger.info("Input args: %s", vars(args))
+
+    logger.info("Opening overrides toml file: %s", args.override_file)
+    override_dict = load_toml_file(args.override_file)
+
+    logger.info("Opening toml file: %s", args.toml_file)
+    toml_dict = load_toml_file(args.toml_file)
+
+    logger.info("Applying overrides to toml file")
+    new_toml_dict = edit_toml(override_dict, toml_dict)
+
+    if args.output_file is not None:
+        logger.info("Generating toml file to: %s", args.output_file)
+        write_toml(args.output_file, new_toml_dict)
+    else:
+        pp.pprint(new_toml_dict)
+
+    logger.info("Finished!")
+
+
+def logger_setup(level):
+    """Sets up logger based on based in log level."""
+    level = level.lower()
+
+    if level == "debug":
+        log_format = (
+            "[%(asctime)s] %(levelname)s [%(name)s.%(funcName)s:%(lineno)d] %(message)s"
+        )
+    else:
+        log_format = "[%(asctime)s] %(levelname)s %(message)s"
+
+    levels = {"info": logging.INFO, "debug": logging.DEBUG, "warning": logging.WARNING}
+    level = levels.get(level)
+
+    logging.basicConfig(
+        stream=sys.stdout, level=level, format=log_format, datefmt="%d-%b-%Y %H:%M:%S"
+    )
+    return logging.getLogger(__name__)
+
+
+def parse_args():
+    """Sets up arguements from CLI input."""
+    parser = argparse.ArgumentParser(
+        description="A script to generate casper-node configs with overridden values"
+    )
+    parser.add_argument("-f", "--toml_file", help="Path to toml file", required=True)
+    parser.add_argument(
+        "-o", "--override_file", help="Path to override file", required=True
+    )
+    parser.add_argument(
+        "-l",
+        "--log_level",
+        help="Sets log level. (default: info)",
+        required=False,
+        default="info",
+        choices=["info", "debug", "warning"],
```

Comment:
We do have some trace-level log statements in the code, too; not sure how useful they are.

Reply:
This is log level for the python script